packages/lib/modules/pool/actions/initialize/useSignPermit2InitializeStep.tsx

```diff
@@ -0,0 +1,47 @@
+import { PublicWalletClient, type InitPoolInputV3, balancerV3Contracts } from '@balancer/sdk'
+import { signPermit2Init } from '@repo/lib/modules/tokens/approvals/permit2/signPermit2Init'
+import { SignPermit2Fn as SignPermit2Fn } from '@repo/lib/modules/tokens/approvals/permit2/useSignPermit2'
+import { useSignPermit2Step } from '@repo/lib/modules/transactions/transaction-steps/useSignPermit2Step'
+import { useUserAccount } from '@repo/lib/modules/web3/UserAccountProvider'
+import { useTokenMetadata } from '@repo/lib/modules/tokens/useTokenMetadata'
+import { GqlChain } from '@repo/lib/shared/services/api/generated/graphql'
+
+type Props = {
+  initPoolInput: InitPoolInputV3
+  chain: GqlChain
+}
+
+export function useSignPermit2InitializeStep({ initPoolInput, chain }: Props) {
+  const { wethIsEth, amountsIn, chainId } = initPoolInput
+
+  const { userAddress } = useUserAccount()
+
+  const signPermit2Fn: SignPermit2Fn = (sdkClient: PublicWalletClient) => {
+    return signPermit2Init({
+      sdkClient,
+      account: userAddress,
+      initPoolInput,
+    })
+  }
+
+  // TODO: figure out better way to get token symbols lol
+  const { symbol: symbolToken0 } = useTokenMetadata(amountsIn?.[0]?.address, chain)
```

Comment:
If you just want the symbol, you can probably use `getToken` from `useTokens`

Reply:
You are probably right, then I don't think we have a better option. At least this data would probably be already cached (for the launch token) if the user has already followed all the other steps. At most, we can use `useTokenMetadata` for the launch token and `getToken` for the collateral one.